### PR TITLE
Add RoomStoreApplyingReadStateUpdatedEventTest

### PR DESCRIPTION
**Pull request by marekoid**

### CODE REVIEW

File: chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                                    unreadCount = 1,
+                                    cursor = null
+                            )
+                        )
+                )
+            }
+
+            it("then the result indicates that the room has been updated") {
+                val roomUpdated: UserInternalEvent.RoomUpdated = result.assertTypedSingletonList()
+                assertThat(roomUpdated.room.unreadCount).isEqualTo(1)
```
Comment (mdpye) [inline chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt]: Is it worth asserting the room ID here, to be clear that the correct room has been updated?

File: chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    describe("given RoomStore with a Room with unreadCount of 0") {
+        val initialRoom = simpleRoom("roomId1", "General")
+        val subject by memoized {
+            RoomStore().apply {
+                initialiseContents(
+                        listOf(initialRoom),
+                        listOf(RoomMembershipApiType("roomId1", listOf("alice"))),
+                        listOf(RoomReadStateApiType("roomId1", 0, null))
+                )
```
Comment (mdpye) [inline chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt]: While I understand keeping test cases simple for understandability, I prefer tests which test a single isolated change in a slightly better populated fixture.

Code which took in this event and emitted an event for a random room from the store would still pass all the tests below, as would code which emitted events for *all* rooms in the store. Mixing a little irrelevant data in to the test fixture is a practical way to get a lot more coverage when testing collections.

You could also test the non-effect on other rooms explicitly, but that would still require additional rooms to exist.

Comment (mdpye) [reply chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt]: I tend to consider my test fixture data in terms of "what's the most wildly incorrect implementation which is not detectable by anything running in this fixture?"

Comment (marekoid) [reply chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt]: Good point! Addressed: d445eab6459ea8eff74c1ba8134caf13ce1a5739

**Comment by marekoid** [comment]
> Would be useful to fill out the description of the PR for posterity. I like the "what & why" format e.g. what does this PR do, and why is it here now

For more complex PR I tend to provide a high-level description if it makes sense and aids the review. This one is trivial and it's clear that it adds a test ("what") and it should be obvious "why" we add tests.

It may be also worth adding that first of all, a log message for a particular commit should address such questions if it may be not clear from the diff itself. PR description come in useful when there's many commits and a summary to aid the review could be useful.

File: chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt (hunk 1 above)
Comment (marekoid) [reply chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomStoreApplyingReadStateUpdatedEventTest.kt]: [Addressed](https://github.com/pusher/chatkit-android/commit/d445eab6459ea8eff74c1ba8134caf13ce1a5739)

**Pull request merged**